<a href="https://colab.research.google.com/github/Fabriciome/data-manipulation-exercises/blob/master/Resolu%C3%A7%C3%A3o_do_Exercicio_unidade_5_Manipulacao_de_Dados_Parte_01.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

![](https://i.imgur.com/JsMvGnP.png)


## Exercícios de manipulação de dados - Parte 1

Neste Jupyter notebook você irá resolver uma exercícios utilizando a biblioteca Pandas.\
Todos os datasets utilizados nos exercícios estão salvos na pasta *datasets*.\
Todo o seu código deve ser executado neste Jupyter Notebook. Por fim, se desejar, revise as respostas com o seu mentor.

In [78]:
import pandas as pd

#### Passo 1. Importando os dados

Carregue os dados salvos no arquivo ***datasets/users_dataset.csv***.\
Esse arquivo possui um conjunto de dados de trabalhadores com 5 colunas separadas pelo símbolo "|" (pipe) e 943 linhas.

Dica: não se esqueça do argumento *sep* quando importar os dados.

In [79]:
users = pd.read_csv('/content/users_dataset.csv',sep='|')

#### Passo 2. Mostre as 15 primeiras linhas do dataset.

*Dica: utilize a função head do DataFrame*

In [80]:
users.head(15)

user_id  age gender     occupation zip_code
0         1   24      M     technician    85711
1         2   53      F          other    94043
2         3   23      M         writer    32067
3         4   24      M     technician    43537
4         5   33      F          other    15213
5         6   42      M      executive    98101
6         7   57      M  administrator    91344
7         8   36      M  administrator    05201
8         9   29      M        student    01002
9        10   53      M         lawyer    90703
10       11   39      F          other    30329
11       12   28      F          other    06405
12       13   47      M       educator    29206
13       14   45      M      scientist    55106
14       15   49      F       educator    97301

#### Passo 3. Mostre as 10 últimas linhas do dataset

In [81]:
users.tail(10)

user_id  age gender     occupation zip_code
933      934   61      M       engineer    22902
934      935   42      M         doctor    66221
935      936   24      M          other    32789
936      937   48      M       educator    98072
937      938   38      F     technician    55038
938      939   26      F        student    33319
939      940   32      M  administrator    02215
940      941   20      M        student    97229
941      942   48      F      librarian    78209
942      943   22      M        student    77841

#### Passo 4. Qual o número de linhas e colunas do DataFrame?

In [82]:
users.shape

(943, 5)

Esse data frame tem 943 linhas e 5 colunas.

#### Passo 5. Mostre o nome de todas as colunas.

In [83]:
users.columns

Index(['user_id', 'age', 'gender', 'occupation', 'zip_code'], dtype='object')

#### Passo 6. Qual o tipo de dado de cada columa?

In [84]:
users.dtypes

user_id        int64
age            int64
gender        object
occupation    object
zip_code      object
dtype: object

#### Passo 7. Mostre os dados da coluna *occupation*.

In [85]:
users.occupation

0         technician
1              other
2             writer
3         technician
4              other
           ...      
938          student
939    administrator
940          student
941        librarian
942          student
Name: occupation, Length: 943, dtype: object

#### Passo 8. Quantas ocupações diferentes existem neste dataset?

In [86]:
ocupacoes = users.occupation.drop_duplicates()
ocupacoes

0         technician
1              other
2             writer
5          executive
6      administrator
8            student
9             lawyer
12          educator
13         scientist
15     entertainment
16        programmer
18         librarian
19         homemaker
22            artist
24          engineer
45         marketing
56              none
59        healthcare
105          retired
111         salesman
137           doctor
Name: occupation, dtype: object

In [87]:
ocupacoes.drop(56)

0         technician
1              other
2             writer
5          executive
6      administrator
8            student
9             lawyer
12          educator
13         scientist
15     entertainment
16        programmer
18         librarian
19         homemaker
22            artist
24          engineer
45         marketing
59        healthcare
105          retired
111         salesman
137           doctor
Name: occupation, dtype: object

In [88]:
ocupacoes.count()

21

#### Passo 9. Qual a ocupação mais frequente?

In [89]:
users.occupation.value_counts()

student          196
other            105
educator          95
administrator     79
engineer          67
programmer        66
librarian         51
writer            45
executive         32
scientist         31
artist            28
technician        27
marketing         26
entertainment     18
healthcare        16
retired           14
salesman          12
lawyer            12
none               9
doctor             7
homemaker          7
Name: occupation, dtype: int64

A ocupação mais frequente é a de estudante que apareceu 196 vezes.

#### Passo 10. Qual a idade média dos usuários?

In [90]:
round(users.age.mean())

34

A idade média dos usuários é 34 anos.

#### Passo 11. Utilize o método describe() para obter diversas informações a respeito do DataFrame.

In [91]:
round(users.age.describe(),1)

count    943.0
mean      34.1
std       12.2
min        7.0
25%       25.0
50%       31.0
75%       43.0
max       73.0
Name: age, dtype: float64

Através do describe na coluna de idade, podemos verificar que além da idade média dos 943 usuários ser 34 anos, a idade mínima é 7 anos e a idade máxima dos usuários é de 73. Também pode-se verificar que 25% dos usuários têm até 25 anos, metade dos usuários têm até 31 anos e que 75% dos usuários tê até 73 anos.

In [103]:
users.gender.describe()

count     943
unique      2
top         M
freq      670
Name: gender, dtype: object

Através do describe na coluna do gênero podemos verificar que dos 943 usuários, 670 são do sexo masculino.

In [102]:
count_total = users.gender.count()
count_M, count_F = users.gender.value_counts()
percentM = round(count_M/count_total*100)
percentF = round(count_F/count_total*100)
print('usuários masculinos =',count_M,'ou',percentM,'%')
print('usuários femininos =',count_F,'ou',percentF,'%')

usuários masculinos = 670 ou 71 %
usuários femininos = 273 ou 29 %


**Parabéns! Você chegou ao fim!**